### PR TITLE
feat: add loading prop to all data-table variants

### LINKED ISSUE #16811: [Feature Request] :loading property for v-data-table
### Problem to solve
The `loading` property is available on the `v-data-table-server`, and disables the table along with showing an indeterminate loading animation. This might also be handy for the `v-data-table`, since there might be situations where we don't need serverside pagination, but still have a loading event on the table. Examples for those might be cases where we load all data at once from the server, but have a button to fetch new data.

### Proposed solution
Make the `loading` and `loading-text` properties and styling available to the default `v-data-table`.

### PR DESCRIPTION
**Pull request by reneroth**
First contribution attempt, please let me know if anything is missing and I'll fix it!  
Resolves #16811

## Description
This adds the `:loading` property to all types of VDataTable, not just VDataTableServer.

## Markup:
```vue
<template>
  <v-app>
    <h1>loading</h1>
    <v-data-table
      :headers="[{title: 'ID', id: 'id'}, {title: 'Username', id: 'name'}]"
      :items="[{id: '1', name: 'John'}, {id: '2', name: 'Jane'}]"
      :loading="true"
    />
  </v-app>
</template>

<script lang="ts"></script>
```

### CODE REVIEW

File: packages/vuetify/src/labs/VDataTable/VDataTableVirtual.tsx [hunk 1]
```diff
@@ -114,6 +116,7 @@ export const VDataTableVirtual = genericComponent<VDataTableVirtualSlots>()({
         class="v-data-table"
         style={{
           '--v-table-row-height': convertToUnit(itemHeight.value),
+          'v-data-table--loading': props.loading,
```
Comment (nekosaur) [inline packages/vuetify/src/labs/VDataTable/VDataTableVirtual.tsx:119]: It's a class, not a style

Comment (reneroth) [reply packages/vuetify/src/labs/VDataTable/VDataTableVirtual.tsx]: Ugh you're right of course. I hope commit-amending and force-pushing is the correct way to go? To abide with the commit rules of the contributor guidelines.

**Review by nekosaur — APPROVED** [review]

**Comment by LoicMoerman** [comment]
Hi, is this ready to be deployed ? Thanks @reneroth and @nekosaur for your work

**Comment by nekosaur** [comment]
@reneroth fix the conflict and I'll approve

**Comment by reneroth** [comment]
> @reneroth fix the conflict and I'll approve

done :)

**Comment by webdevnerdstuff** [comment]
@nekosaur Can we please get this reviewed so it can go out with the next release?

**Comment by LoicMoerman** [comment]
Hey, is this ready to get review ? Thanks @nekosaur

**Review by nekosaur — DISMISSED** [review]

**Comment by nekosaur** [comment]
hmm, that's not how I resolved that? 🤔

**Review by nekosaur — APPROVED** [review]

**Comment by reneroth** [comment]
thank you nekosaur, very proud to have been allowed to make a (tiny) commit to this great library 💕

**Pull request merged**